src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -649,6 +679,95 @@ def get_observation(self, event: Event) -> pd.DataFrame:
         for col in ["employer_address_id", "employer_name"]:
             df_w2[col] = df_w2["employer_id"].map(emp[col])
 
+        # re-initialize income-to-date series for next year of income counting
+        self.income_last_year = self.income_this_year
+        self.income_this_year = empty_income_series()
+
+        return df_w2[self.output_columns]
+
+
+class TaxDependentsObserver(BaseObserver):
+    """Class for observing columns relevant to identifying dependents in tax data.
+
+    This is most important for the 1040 data, but relies on data from
+    the W2 observer, and is better represented in a "long" form with a
+    row for each guardiant/dependent pair
+
+    """
+
+    INPUT_VALUES = ["household_details"]
+    ADDITIONAL_INPUT_COLUMNS = [
+        "alive",
+        "in_united_states",
+        "tracked",
+        "ssn",
+    ]
+    OUTPUT_COLUMNS = [
+        "guardian_id",
+        "dependent_id",
+        "first_name_id",
+        "middle_name_id",
+        "last_name_id",
+        "age",
+        "date_of_birth",
+        "sex",
+        "ssn",
+        "address_id",
+        "tax_year",
+    ]
+
+    def __init__(self, w2_observer):
+        super().__init__()
+        self.w2_observer = w2_observer
+
+    def __repr__(self):
+        return f"TaxDependentsObserver()"
+
+    @property
+    def name(self):
+        return f"tax_dependents_observer"
+
+    @property
+    def input_columns(self):
+        return self.DEFAULT_INPUT_COLUMNS + self.ADDITIONAL_INPUT_COLUMNS
+
+    @property
+    def input_values(self):
+        return self.INPUT_VALUES
+
+    @property
+    def output_columns(self):
+        return self.OUTPUT_COLUMNS
+
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        self.clock = builder.time.clock()
+
+    def to_observe(self, event: Event) -> bool:
+        """Observe if Jan 1 falls during this time step"""
+        tax_date = dt.datetime(event.time.year, 1, 1)
+        return self.clock() < tax_date <= event.time
+
+    def get_observation(self, event: Event) -> pd.DataFrame:
+        pop_full = self.population_view.get(event.index)
+        household_details = self.pipelines["household_details"](pop_full.index)
+        pop_full[["address_id", "housing_type"]] = household_details[
+            ["address_id", "housing_type"]
+        ]
+
+        # Tracked, US population to be dependents
+        pop = pop_full[
+            (
+                pop_full["alive"] == "alive"
+            )  # really should include people who died in last year
+            & pop_full["tracked"]  # ??? should we include untracked, too?
```

Comment:
The only way a simulant can become untracked is by emigrating so `in_united_states` and `tracked` are identical, but I think it's good to check both.